control_protocol.md

```diff
@@ -330,17 +330,490 @@ A Component MUST also offer a list of all possibly callable methods in accordanc
+
+::: json
+{
+    "openrpc": "1.2.6",
+    "info": {
+        "title": "Polling Actor",
+        "version": "0.1.0"
+    },
+    "methods": [
+        {
+            "name": "start_polling",
+            "params": [
+                {
+                    "name": "polling_interval",
+                    "schema": {
+                        "anyOf": [
+                            {
+                                "type": "number"
+                            },
+                            {
+                                "type": "null"
+                            }
+                        ]
+                    },
+                    "required": false
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "stop_polling",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "get_polling_interval",
+            "params": [],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "number"
+                },
+                "required": true
+            }
+        },
+        {
+            "name": "set_polling_interval",
+            "params": [
+                {
+                    "name": "polling_interval",
+                    "schema": {
+                        "type": "number"
+                    },
+                    "required": true
+                }
+            ],
+            "result": {
+                "name": "result",
+                "schema": {
+                    "type": "null"
+                },
+                "required": true
+            }
+        },
+    ],
+    "components": {
+        "schemas": {}
+    }
+}
+:::
+
 
-### Messages for Transport Layer
+#### Locking Actor
 
-- SIGNIN
-- SIGNOUT
-- ACKNOWLEDGE
-- ERROR
-- PING
-- CO_SIGNIN
-- CO_SIGNOUT
+An {ref}`control_protocol.md#Actor` which support locking resources MUST offer the following methods.
 
 :::{note}
-TODO How to make these messages work? Define them directly in the transport layer?
+TBD change the python code to json
 :::
+
+::: python
+    def lock(self, resource: Optional[str] = None) -> bool: ...
+
+    def unlock(self, resource: Optional[str] = None) -> None: ...
+
+    def force_unlock(self, resource: Optional[str] = None) -> None: ...
+:::
+
+
+### Errors
+
+Every error has a code and a message.
+Additionally they may have a ``data`` field with more information.
+
+According to JSONRPC, applications can define error codes between -32000 and -32099.
```

Comment:
100 error codes is not terrribly much :-/

Reply:
Especially, as there are only a handful of [error codes](https://www.jsonrpc.org/specification#error_object) defined:

<html><body>
<!--StartFragment-->

code | message | meaning
-- | -- | --
-32700 | Parse error | Invalid JSON was received by the server.An error occurred on the server while parsing the JSON text.
-32600 | Invalid Request | The JSON sent is not a valid Request object.
-32601 | Method not found | The method does not exist / is not available.
-32602 | Invalid params | Invalid method parameter(s).
-32603 | Internal error | Internal JSON-RPC error.
-32000 to -32099 | Server error | Reserved for implementation-defined server-errors.

<!--EndFragment-->
</body>
</html>